src/Core/WinSWCore/Download.cs

```diff
@@ -29,16 +29,21 @@ public enum AuthType
 
         private static readonly ILog Logger = LogManager.GetLogger(typeof(Download));
 
-        public readonly string From;
-        public readonly string To;
-        public readonly AuthType Auth;
-        public readonly string? Username;
-        public readonly string? Password;
-        public readonly bool UnsecureAuth;
-        public readonly bool FailOnError;
-        public readonly string? Proxy;
+        public string from;

```

Comment:
Would be great to apply the same approach as in Log / YAML Descriptor classes. using public fields is an obstacle for future API changes. Not a blocker for the prototype

Reply:
Updated this and pushed.